### DISCUSSION TITLE
Credentials SignIn

### DISCUSSION THREAD

**Discussion by TheDarkSword** [Hebilicious/authjs-nuxt#7] [Q&A] [answered]
Hello, I have added the SignIn credentials, but I don't want to be redirected if the credentials are incorrect. I need a boolean or something similar to determine if the SignIn failed, so that I can display an appropriate message.

By using the signIn(credentials, { redirect: false, username: 'admin', password: 'admin' }) syntax, I am able to disable the redirect. However, I'm not aware of any method to directly retrieve the login status in case of failure. Is there any alternative method to achieve this?

**Comment by Hebilicious** [comment]
Hi,
The sign-in helper is defined [here](https://github.com/Hebilicious/authjs-nuxt/blob/366a0a46aed5be245a30d02f25e8a7179f7bd319/packages/authjs-nuxt/src/runtime/composables/useAuth.ts#L32).
If it fails, your status will be error and an error will be thrown. 

What would you like to happen ?

**Comment by TheDarkSword** [comment]
Hi,
I attempted to test the following code for verification purposes:
```
try {
    signIn(`credentials`, { redirect: false, username: 'admin', password: 'admins' })
    console.log("SignIn")
  } catch (error) {
    console.log("Error SignIn: " + error)
  }
```
However, it only outputs "SignIn" and does not enter the catch block. The "async authorize(..." function in the CredentialsProvider returns null when the username and password are not "admin,". I am confident that I have made a mistake in my implementation. If you could kindly assist me, I would greatly appreciate it.
```
async authorize(credentials, req) {
    if(credentials.username === "admin" && credentials.password === "admin") {
        return { id: "1", name: "admin", email: "admin" }
    }
    return null
}
```

**Reply by Hebilicious** [reply]
- signIn is async, you need to await it.
- I haven't tried all auth.js providers, but I assume they work, still make sure that you use the `@auth/core` [providers](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/credentials.ts) and not the next-auth ones, and configure them properly
- You can import `status` from useAuth to check if it's correctly goes to error 

If you find out there's a bug, please open an issue with a stackblitz so I can investigate further.

**Reply by TheDarkSword** [reply]
I imported status and it still on "loading".
The provider I'm usign is: `import CredentialsProvider from "@auth/core/providers/credentials"`
I added await before signIn method but nothing changed.

P.S. If the credentials are correct all goes perfectly.

**Comment by TheDarkSword** [comment]
https://stackblitz.com/edit/nuxt-starter-3bmqov?file=README.md.

This is an illustrative example.

P.S. The website's behavior aligns with the description provided. However, when attempting to log in with the correct credentials on StackBlitz, an error occurs. I attempted to use the @auth library in Next.js, but encountered the same issue, which seems to be specific to StackBlitz.

**Reply by Hebilicious** [reply]
Thank you, I will investigate. If you could open an issue that would be ideal 👍🏽

**Reply by Hebilicious** [reply]
I've added this feature too in the latest version.

**Comment by TheDarkSword** [comment]
Additionally, in the event that I attempt to access a webpage that necessitates authentication, I will be automatically redirected to the default page, which is set as "/". Is it possible to designate a custom page for redirection? If so, how can I do?

EDIT: I saw in the code that is not configurable yet

**Reply by Hebilicious** [reply]
That should be easy to add, open a PR, or open a feature request issue and I'll add it in.

**Reply by Hebilicious** [reply]
Actually I just added this to the latest version :

```
    authJs: {
       guestRedirectTo: "/redirect",
    }
  ```
  So you can customise it.

**Comment by Hebilicious** [comment, marked as answer]
Use v0.1.0 https://github.com/Hebilicious/authjs-nuxt/releases/tag/v0.1.0
